File: modules/spring-web/testdata/java/inspection/duplicatedEndpoint/src/com/DuplicatedEndpointController.java
package com.example.springcoredemo.web;

import org.springframework.web.bind.annotation.*;

import java.util.Map;
import java.util.Optional;

@RestController
public class DuplicatedEndpointController {
    
    @RequestMapping(path = "/same/uri", method = {RequestMethod.GET, RequestMethod.PUT})
    @ResponseBody
    public String conflictedTwice() {
        return "conflictedTwice";
    }
    
    @GetMapping("/same/uri")
    @ResponseBody
    public String conflictedGet() {
        return "conflictedGet";
    }
    
    @PutMapping("/same/uri")
    @ResponseBody
    public String conflictedPut() {
        return "conflictedPut";
    }
    
    @PostMapping("/same/uri")
    @ResponseBody
    public String postWithoutConflict() {
        return "postWithoutConflict";
    }
    
}

File: modules/spring-web/testdata/kotlin/inspection/duplicatedEndpoint/src/com/DuplicatedEndpointController.kt
package com.example.springcoredemo.web

import org.springframework.web.bind.annotation.*

@RestController
class DuplicatedEndpointController {

    @RequestMapping(path = "/same/uri", method = RequestMethod.GET)
    @ResponseBody
    fun conflictedRequestMapping(): String {
        return "conflictedRequestMapping"
    }

    @GetMapping("/same/uri")
    @ResponseBody
    fun conflictedGet(): String {
        return "conflictedGet"
    }

    @PostMapping("/same/uri")
    @ResponseBody
    fun postWithoutConflict(): String {
        return "postWithoutConflict"
    }

}